core/fetcher.go

```diff
@@ -82,13 +84,20 @@ func (f *BlockFetcher) SubscribeNewBlockEvent(ctx context.Context) (<-chan *bloc
 
 // UnsubscribeNewBlockEvent stops the subscription to new block events from Core.
 func (f *BlockFetcher) UnsubscribeNewBlockEvent(ctx context.Context) error {
-	// close the new block channel
 	if f.newBlockCh == nil {
 		return fmt.Errorf("no new block event channel found")
 	}
+	if f.doneCh == nil {
+		return fmt.Errorf("no stop signal chan found in fetcher")
+	}
 	defer func() {
+		// send stop signal
+		f.doneCh <- struct{}{}
```

Comment:
This part can be completely removed. Golang's `close` sends nil by default and you call it on line 98. By sending an additional empty struct you signal to a reader twice.